flink-metrics/flink-metrics-core/src/main/java/org/apache/flink/metrics/groups/OperatorIOMetricGroup.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.metrics.groups;
+
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.MetricGroup;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Metric group that contains shareable pre-defined IO-related metrics for operators.
+ *
+ * <p>All metrics can only be accessed in the main operator thread.
```

Comment:
This is a bit misleading for internal usages because we do access metrics outside the operator thread, from the reporter thread.
I also wonder whether "can" is the right word; you "can" totally access them from a different thread. You _shouldn't_ though (in general).

Reply:
I replaced with `You should only update the metrics in the main operator thread.`